heartbeat/beater/heartbeat.go

```diff
@@ -66,11 +68,20 @@ func New(b *beat.Beat, rawConfig *conf.C) (beat.Beater, error) {
 		return nil, fmt.Errorf("error reading config file: %w", err)
 	}
 
+	// Check if any of these can prevent using states client
 	stateLoader, replaceStateLoader := monitorstate.AtomicStateLoader(monitorstate.NilStateLoader)
 	if b.Config.Output.Name() == "elasticsearch" && !b.Manager.Enabled() {
 		// Connect to ES and setup the State loader if the output is not managed by agent
-		if err := makeStatesClient(b.Config.Output.Config(), replaceStateLoader, parsedConfig.RunFrom); err != nil {
-			logp.L().Warnf("could not connect to ES for state management during initial load: %s", err)
+		// Note this, intentionally, blocks until connected or max attempts reached
+		esClient, err := makeESClient(b.Config.Output.Config(), 3, 2*time.Second)
+		if err != nil {
+			if parsedConfig.RunOnce {
+				return nil, fmt.Errorf("run_once mode fatal error: %w", err)
+			} else {
+				logp.L().Warn(err)
```

Comment:
this causes the error to be chatty as its already logged by libbbeat, better to ignore it.

Reply:
++. IMO, there should still be a warning specifying that states client could not connect and therefore will not be used, so I refactored the error message to reduce chattiness,  wdyt?